packages/elements-dev-portal/src/components/NodeContent/NodeContent.tsx

```diff
@@ -130,13 +130,22 @@ const externalRegex = new RegExp('^(?:[a-z]+:)?//', 'i');
 const LinkComponent: CustomComponentMapping['a'] = ({ children, href, title }) => {
   const ctx = React.useContext(NodeLinkContext);
 
-  if (href && externalRegex.test(href)) {
-    // Open external URL in a new tab
-    return (
-      <a href={href} target="_blank" rel="noreferrer" title={title ? title : undefined}>
-        {children}
-      </a>
-    );
+  try {
+    if (href && externalRegex.test(href)) {
+      const baseURL = window.location.host;
+      const hrefURL = new URL(href).host;
+
+      if (baseURL === hrefURL) {
+        // Open URL in same tab if domain match
+        return (
+          <a href={href} rel="noreferrer" title={title ? title : undefined}>
+            {children}
+          </a>
+        );
```

Comment:
i would prefer the logic to stay in the order it is and not get inverted

also, why `noreferrer` when it's opening in the same tab?

Reply:
@ben-smartbear , made changes according to review comment, could you please re-review?